core/src/main/java/org/apache/calcite/rel/hint/HintStrategyOr.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rel.hint;
+
+import org.apache.calcite.rel.RelNode;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Strategy to decide if a hint should apply to a relational expression
+ * by using a series of {@link HintStrategy} rules in a given order.
+ * Returns true if any rule satisfies, otherwise returns false.
+ */
+public class HintStrategyOr implements HintStrategy {
+  //~ Instance fields --------------------------------------------------------
+
+  private ImmutableList<HintStrategy> strategies;
+
```

Comment:
We can merge this class with `HintStrategyCascade`, you can reference `CompositeOperandTypeChecker`.

Reply:
Make sense.  👍